src/catalua.cpp

```diff
@@ -387,15 +387,17 @@ class LuaValue
                 lua_setglobal( L, global_name );
             }
         }
-        static void push( lua_State *const L, const cata::optional<T> &value ) {
+        template<typename U>
+        static void push( lua_State *const L, const cata::optional<U> &value ) {
             if( value ) {
                 push( L, *value );
             } else {
                 lua_pushnil( L );
             }
         }
-        template<typename ...Args>
-        static void push( lua_State *const L, Args &&... args ) {
+        template<typename ...Args,
+                 typename std::enable_if<std::is_constructible<T, Args...>::value, int>::type = 0 >
```

Comment:
```suggestion
                 typename std::enable_if<std::is_constructible<T, Args...>::value, int>::type* = 0 >
```
I'm surprised if this works; `::type` will be void when it is defined, to which you cannot assign 0.  I think the normal pattern is as in this suggested change.